app/controllers/concerns/versionable.rb

```
@@ -0,0 +1,75 @@
+module Versionable
+  extend ActiveSupport::Concern
+
+  def get_modifiable(obj)
+    if obj.respond_to? 'template'
+      template = obj.template
+    elsif obj.is_a? Template
+      template = obj
+    else
+      return nil # Throw error as wrong obj added
```

remember to raise error instead of returning nil